core/src/main/java/com/yahoo/ycsb/measurements/OneMeasurementRaw.java

```diff
@@ -0,0 +1,192 @@
+import java.io.IOException;
+import java.io.PrintStream;
+import java.util.Properties;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import com.yahoo.ycsb.measurements.exporter.MeasurementsExporter;
+
+/**
+ * Record a series of measurements as raw data points without down sampling,
+ * optionally write to an output file when configured.
+ *
+ * @author stfeng
+ *
+ */
+public class OneMeasurementRaw extends OneMeasurement {
+  /**
+   * One raw data point, two fields: timestamp (ms) when the datapoint is
+   * inserted, and the value.
+   */
+  class RawDataPoint {
+    private final long timestamp;
+    private final int value;
+
+    public RawDataPoint(int value) {
+      this.timestamp = System.currentTimeMillis();
+      this.value = value;
+    }
+
+    public long timeStamp() {
+      return timestamp;
+    }
+
+    public int value() {
+      return value;
+    }
+  }
+
+  class RawDataPointComparator implements Comparator<RawDataPoint> {
+    @Override
+    public int compare(RawDataPoint p1, RawDataPoint p2){
+      if (p1.value() < p2.value()){
+        return -1;
+      } else if (p1.value() == p2.value()) {
+        return 0;
+      } else {
+        return 1;
+      }
+    }
+  }
+
+  /**
+   * Optionally, user can configure an output file to save the raw data points.
+   * Default is none, raw results will be written to stdout.
+   *
+   */
+  public static final String OUTPUT_FILE_PATH = "measurement.raw.output_file";
+  public static final String OUTPUT_FILE_PATH_DEFAULT = "";
+  private String outputFilePath = "";
+  private final PrintStream outputStream;
+
+  private ArrayList<RawDataPoint> measurements;
+  private long totalLatency = 0;
+
+  // A window of stats to print summary for at the next getSummary() call.
+  // It's supposed to be a one line summary, so we will just print count and 
+  // average.
+  private int windowOperations = 0;
+  private long windowTotalLatency = 0;
+
+  public OneMeasurementRaw(String name, Properties props) {
+    super(name);
+
+    outputFilePath = props.getProperty(OUTPUT_FILE_PATH,
+        OUTPUT_FILE_PATH_DEFAULT);
+    if (!outputFilePath.isEmpty()) {
+      System.out.println("Raw data measurement: will output to result file: " +
+          outputFilePath);
+
+      try {
+        outputStream = new PrintStream(
+            new FileOutputStream(outputFilePath, true),
+            true);
+      } catch (FileNotFoundException e) {
+        throw new RuntimeException("Failed to open raw data output file", e);
+      }
+
+    } else{
+      System.out.println("Raw data measurement: will output to stdout.");
+      outputStream = System.out;
+
+    }
+    measurements = new ArrayList<RawDataPoint>(1000);
+  }
+
+  @Override
+  public synchronized void measure(int latency) {
+    totalLatency += latency;
+    windowTotalLatency += latency;
+    windowOperations++;
+
+    measurements.add(new RawDataPoint(latency));
```

Comment:
This is protected because the method is marked as "synchronized". (this is the same approach taken by other measurement type classes)